src/text/text.rs

```diff
@@ -632,12 +630,11 @@ impl<T: fmt::Display> ToText for T {
 
 impl fmt::Display for Text<'_> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        for (position, line) in self.iter().with_position() {
-            if position == Position::Last || position == Position::Only {
-                write!(f, "{line}")?;
-            } else {
+        if let Some((last, rest)) = self.lines.split_last() {
+            for line in rest {
                 writeln!(f, "{line}")?;
             }
+            last.fmt(f)?;
```

Comment:
I'm curious about what the difference is here over would be compared to calling `write!(f, {last})?;` I suspect that would be clearer.